Makefile

```
@@ -0,0 +1,105 @@
+include .bingo/Variables.mk
+FILES_TO_FMT      	 ?= $(shell find . -path ./vendor -prune -o -name '*.go' -print)
+MDOX_VALIDATE_CONFIG ?= .mdox.validate.yaml
+
+GO111MODULE       ?= on
+export GO111MODULE
+
+GOBIN ?= $(firstword $(subst :, ,${GOPATH}))/bin
+
+# Tools.
+GIT ?= $(shell which git)
+
+# Support gsed on OSX (installed via brew), falling back to sed. On Linux
+# systems gsed won't be installed, so will use sed as expected.
+SED ?= $(shell which gsed 2>/dev/null || which sed)
+
+define require_clean_work_tree
+	@git update-index -q --ignore-submodules --refresh
+
+    @if ! git diff-files --quiet --ignore-submodules --; then \
+        echo >&2 "$1: you have unstaged changes."; \
+        git diff-files --name-status -r --ignore-submodules -- >&2; \
+        echo >&2 "Please commit or stash them."; \
+        exit 1; \
+    fi
+
+    @if ! git diff-index --cached --quiet HEAD --ignore-submodules --; then \
+        echo >&2 "$1: your index contains uncommitted changes."; \
+        git diff-index --cached --name-status -r --ignore-submodules HEAD -- >&2; \
+        echo >&2 "Please commit or stash them."; \
+        exit 1; \
+    fi
+
+endef
+
+help: ## Displays help.
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n\nTargets:\n"} /^[a-zA-Z_-]+:.*?##/ { printf "  \033[36m%-10s\033[0m %s\n", $$1, $$2 }' $(MAKEFILE_LIST)
+
+.PHONY: all
+all: format build
+
+.PHONY: build
+build: check-git deps ## Build obsctl.
+	@echo ">> building obsctl"
+	@GOBIN=$(GOBIN) go install github.com/observatorium/obsctl
+
+.PHONY: check-comments
+check-comments: ## Checks Go code comments if they have trailing period (excludes protobuffers and vendor files). Comments with more than 3 spaces at beginning are omitted from the check, example: '//    - foo'.
+	@echo ">> checking Go comments trailing periods\n\n\n"
+	@./scripts/build-check-comments.sh
+
+.PHONY: deps
+deps: ## Ensures fresh go.mod and go.sum.
+	@go mod tidy
+	@go mod verify
+
+.PHONY: docs
+docs: build $(MDOX) ## Generates config snippets and doc formatting.
+	@echo ">> generating docs $(PATH)"
+	PATH=${PATH}:$(GOBIN) $(MDOX) fmt -l --links.validate.config-file=$(MDOX_VALIDATE_CONFIG) *.md
```

```suggestion
	$(MDOX) fmt -l --links.validate.config-file=$(MDOX_VALIDATE_CONFIG) *.md
```

We probably dont need to modify the PATH because $(MDOX) is already an absolute path (`$GOBIN/mdox-$VERSION`)

---

We have this so that we can call other binaries through mdox using `mdox-exec` directives. So, when we paste the output of `obsctl --help` in a file, we need PATH to have GOBIN. You can see a [test run](https://github.com/saswatamcode/obsctl/runs/5199982770?check_suite_focus=true) I did, which fails when this is removed. 🙂